src/main/java/Duke.java

```diff
@@ -1,10 +1,152 @@
+import util.Parser;
+import util.Storage;
+import util.Ui;
+
+import task.TaskList;
+import task.Event;
+import task.Deadline;
+import task.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Scanner;
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    private File newFile;
+    private Parser parser;
+
+
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        newFile = new File(filePath);
+        tasks = new TaskList();
+        parser = new Parser();
+
+        try {
+            if (newFile.exists()) {
+                storage.loadFile(tasks.list);
+            }
+        } catch (FileNotFoundException e) {
+            ui.reply("Please create the text file data/duke.txt");
+        }
+
+
+
+    }
+
+    public void run() {
```

Comment:
Avoid long methods. Try to split this method up into smaller parts.
![image](https://user-images.githubusercontent.com/80502407/151958785-f21590ed-a80b-470a-897a-0e2239eb2e3f.png)